pyproject.toml

```diff
@@ -1,3 +1,3 @@
 [build-system]
-requires = ["setuptools>=40.8.0", "wheel", 'cython>=0.21', 'numpy>=1.6, <2.0']
-build-backend = "setuptools.build_meta:__legacy__"
\ No newline at end of file
+requires = ["setuptools>=40.8.0", "wheel", 'cython>=0.21', 'numpy>=2.0']
```

Comment:
This changes the minimum version requirement from 1.6 to 2.0, rather than allowing 1.6 and upwards (and including 2.0). I don't think that is useful. Would prefer ``numpy>=1.6`` here.

Reply:
Apologies, I should have explained as this is somewhat of a change in numpy 2.0. This doesn't actually change the minimum version required to install the package, only what is used during the build process.

A backwards compatible stable ABI in numpy 2.0, so if you build against it, the packages will still work with numpy 1.X. The reverse is not true, so builds against 1.X won't work with numpy 2.0. 

See e.g. https://numpy.org/devdocs/dev/depending_on_numpy.html#numpy-2-0-specific-advice

Reply:
As you can see from the checks, this does not work. We cannot build ``pypmc`` for python versions that do not have a valid ``numpy>=2.0`` release, which includes the build tuple ``macOS/arm64/cp38``:
```
        ERROR: Ignored the following versions that require a different python version: 1.25.0 Requires-Python >=3.9; 1.25.1 Requires-Python >=3.9; 1.25.2 Requires-Python >=3.9; 1.26.0 Requires-Python <3.13,>=3.9; 1.26.1 Requires-Python <3.13,>=3.9; 1.26.2 Requires-Python >=3.9; 1.26.3 Requires-Python >=3.9; 1.26.4 Requires-Python >=3.9; 2.0.0 Requires-Python >=3.9; 2.0.1 Requires-Python >=3.9; 2.0.2 Requires-Python >=3.9; 2.1.0 Requires-Python >=3.10; 2.1.0rc1 Requires-Python >=3.10; 2.1.1 Requires-Python >=3.10
        ERROR: Could not find a version that satisfies the requirement numpy>=2.0 (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 1.13.3, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5, 1.20.0, 1.20.1, 1.20.2, 1.20.3, 1.21.0, 1.21.1, 1.21.2, 1.21.3, 1.21.4, 1.21.5, 1.21.6, 1.22.0, 1.22.1, 1.22.2, 1.22.3, 1.22.4, 1.23.0, 1.23.1, 1.23.2, 1.23.3, 1.23.4, 1.23.5, 1.24.0, 1.24.1, 1.24.2, 1.24.3, 1.24.4)
        ERROR: No matching distribution found for numpy>=2.0
        [end of output]
```

Reply:
@dvandyk Yes, you are right. Python 3.8 is no longer supported by numpy and so their latest releases don't have builds (hence building against numpy 2.0 is failing). Their schedule is here for reference https://numpy.org/neps/nep-0029-deprecation_policy.html. 

@dvandyk What are your thoughts on 3.8 support? Is that still required or would it be all right to change to >=3.9 (maybe add python 3.12 builds at the same time?). I could make this change in a separate PR if you are amenable. If 3.8 is required, I think it will need some special cases in the build with numpy < 2.0. I think that is possible to do.

Reply:
I think we should drop support for python version 3.8. Can you force push a few commits that
a) drop python 3.8 support
b) require numpy>=2.0 to build
c) add python 3.12 support?
d) release a new pypmc version.

@ahnitz Tracking a) through c) now in PR #108.